docs/changes/units/15642.api.rst

```diff
@@ -0,0 +1 @@
+The deprecated ``Quantity.nansum()`` method has been removed.
```

Comment:
Maybe should mention the alternative that was in the deprecation message?
```suggestion
The deprecated ``Quantity.nansum()`` method has been removed; use ``np.nansum()`` instead.
```

Reply:
Yes, good idea. Just pushed an extra commit making the change, but with `skip ci` to avoid a rerun. Please do squash merge the two commits.